CRS/Cirrus.Node/Controllers/IndexerContractsController.cs

```diff
@@ -79,11 +83,76 @@ public IActionResult GetContractInfo([FromQuery] string txHash)
                 logResponses = deserializer.MapLogResponses(receipt.Logs);
             }
 
-            return this.Json(new ContractReceiptResponse(receipt, logResponses ?? new List<LogResponse>(), this.network)
+            EnrischLogs(receipt, typeName, logResponses);
+
+            return this.Json(new ContractReceiptResponse(receipt, logResponses, this.network)
             {
                 ContractCodeType = typeName,
             });
         }
 
+        private void EnrischLogs(Receipt receipt, string typeName, List<LogResponse> logResponses)
+        {
+            if (typeName == "StandardToken")
+            {
+                if (receipt.Success)
+                {
+                    if (receipt.MethodName == null && receipt.NewContractAddress != null)
+                    {
+                        // this is the constructor we want to fetch the name, symbol, decimals and total supply.
+                        IStateRepositoryRoot stateAtHeight = this.stateRoot.GetSnapshotTo(receipt.PostState.ToBytes());
+
+                        uint160 addressNumeric = receipt.NewContractAddress;
+
+                        byte[] tokenName = stateAtHeight.GetStorageValue(addressNumeric, Encoding.UTF8.GetBytes("Name"));
```

Comment:
Encoding.UTF8 could be replaced with const encoding = Encoding.UTF8 and then referenced all places below. I'm not sure if the compiler actually might optimize this automatically.